### ISSUE TITLE
Creation of a Battle mode to assist in CLI integration

### ISSUE THREAD

**Issue by jintae412** [uchicago-cs/chiventure#1043]
Similar to the mode functionality that the Dialogue team created in this issue: https://github.com/uchicago-cs/chiventure/issues/864 we would like to see a mode for battles as well. In doing so we can limit the acceptable user inputs to those that we specify.

Steps that we need to take would likely include ensuring that battles are accessible by the game context, as well as using the mode functionality provided by the Dialogue team to implement our own mode, based off of a mixture of their work and our currently existing battle_flow_example.

**Comment by Elsa-Athiley** [comment]
We've began to implement Battle mode. So far we were able to add BATTLE as another possible mode enum, created a new example file that will hold our monkey-patched example and added functions that will run the enemies turn and allow the user to consume items in actual battle (previously this only existed in the example created by last years team)

**Comment by jintae412** [comment]
Tentatively the battle mode implementation is done. We had to edit the ctx struct in game-state so that the CLI has a way to access the chiventure_battle_ctx struct and read/influence battles. In addition, changes were made to the battle print functions so that they facilitated printing to the CLI using the CLI print functions. This largely meant for things like printing the list of moves and battle items, to create one string and concatenate all the entries together so that that given string could then be passed as a parameter to the CLI print functions.

In addition, to ensure that the find battle moves and find battle items functions worked with the parsed inputs, we had to add a requirement that the info field for battle moves and the name field for battle items are in all caps. Consequently, there is no support for moves with info fields (names) that are more than one word, although this can easily be added by mimicking what we have done with the battle items.

**Comment by Elsa-Athiley** [comment]
Initially we printed out to the CLI directions for using valid commands as well as the complete list of moves and items. Now we have changed it so 'list moves' and 'list items' are standalone commands that can be used at any point in the battle.

**Comment by Elsa-Athiley** [comment]
closing statement:  To summarize we were able to integrate with CLI enough to create a `battle_cli_example.c` file that runs within the actual chiventure ctx rather than just through the terminal. We utilized the mode module created by the NPC team to introduce a battle mode that allowed us to use our own `run_battle_mode` function that took in user input and completed the correct battle actions. We also changed some battle_flow and battle_print functions to work with the CLI, this just included having them return strings that can then be printed to the CLI. Further work can be done to allow more commands to be used during the battle
